### PR TITLE
Feature/25 delete account data

### PR DESCRIPTION
**Pull request by Atanasov-AA**
In this pr was fixed a keycloak auth bug as well. For more information check the https://stackoverflow.com/a/72333847 link and the application-dev.yml file

### CODE REVIEW

File: am-ui/src/app/components/profile/profile.component.html [hunk 1]
```diff
@@ -40,8 +40,47 @@ <h1>
...
+        </mp-custom-modal>
+      </div>
+    </div>
+
+    <div
+      *ngIf="showDeleteUserAlert"
+      class="result-banner"
+      [ngClass]="deleteUserAlertCssClassesArray"
+    >
+      {{ deleteUserAlertText }}
```
Comment (aschiakros) [inline am-ui/src/app/components/profile/profile.component.html:73]: Is there a | translate missing?

File: am-ui/src/app/components/profile/profile.component.ts [hunk 2]
```diff
@@ -75,6 +81,45 @@ export class ProfileComponent implements OnInit {
       );
   }
 
+  onDeleteProfile() {
+    this.userService.deleteUser(this.userId).subscribe(
+      data => {
+        this.deleteUserAlertText = this.translatePipe.transform(
```
Comment (Atanasov-AA) [inline am-ui/src/app/components/profile/profile.component.ts:87]: @aschiakros  here is the translate

File: am-ui/src/app/components/profile/profile.component.html (hunk 1 above)
Comment (Atanasov-AA) [reply am-ui/src/app/components/profile/profile.component.html:73]: Translate is implemented in the ocde 
https://github.com/AkrosAG/Akros-Marketplace/pull/277/files#r1063529842

File: marketplace-service/src/main/java/ch/akros/marketplace/service/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    .bodyValue(bodyValues)
+                    .retrieve()
+                    .bodyToMono(JSONObject.class)
+                    .block();
+
+            return Optional.of(jsonResponse.get("access_token").toString());
+        } catch (Exception e) {
+            log.error(e.getMessage(), e);
+        }
+        return Optional.empty();
```
Comment (aschiakros) [inline marketplace-service/src/main/java/ch/akros/marketplace/service/service/impl/UserServiceImpl.java:85]: It's more of a question for myself: Is the return Optional.empty(); still reached after an Exception?

**Review by aschiakros — APPROVED** [review]

File: marketplace-service/src/main/java/ch/akros/marketplace/service/service/impl/UserServiceImpl.java (hunk 3 above)
Comment (Atanasov-AA) [reply marketplace-service/src/main/java/ch/akros/marketplace/service/service/impl/UserServiceImpl.java:85]: Yes, if an exception is thrown in the try block, the return in the try block will **not** be executed, the code will go to the catch block and then in the end will return an empty.

File: marketplace-service/src/main/resources/application-prod.yml [hunk 4]
```diff
@@ -11,11 +11,15 @@ spring:
     password: am
   liquibase:
     enabled: false
+  keycloak_client:
+    base_url: * # TODO add client_secret for prod
```
Comment (asiday) [inline marketplace-service/src/main/resources/application-prod.yml:15]: Could you please specify it as a new Ticket.

File: am-ui/src/assets/i18n/en.json [hunk 5]
```diff
@@ -92,6 +93,13 @@
       "email": "E-Mail",
       "phone": "Phone number"
     },
+    "modal": {
+      "title": "Delete profile permanently",
+      "body": "Are you sure you want to delete your profile?"
+    },
+    "deleteProfile": "Delete my profile and all data",
```
Comment (asiday) [inline am-ui/src/assets/i18n/en.json:100]: At the end of some sentences you are using ".", and at another there are no dots. Please use one formatting style.

File: api.yaml [hunk 6]
```diff
@@ -134,6 +134,34 @@ paths:
               schema:
                 $ref: "#/components/schemas/Problem"
 
+  /users/{userId}:
+    delete:
+      tags:
+        - Users
+      summary: Delete an user
```
Comment (asiday) [inline api.yaml:141]: Delete **a** user

**Review by asiday — APPROVED** [review]

File: marketplace-service/src/main/java/ch/akros/marketplace/service/service/impl/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        try {
+            keycloakClient.delete()
+                    .uri(requestUrl)
+                    .header("Authorization", authHeader)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .retrieve()
+                    .bodyToMono(String.class)
+                    .block();
+
+            this.topicService.deleteTopicsForUser(userId);
```
[the remark this answers is not in the document]
Comment (Atanasov-AA) [reply marketplace-service/src/main/java/ch/akros/marketplace/service/service/impl/UserServiceImpl.java:56]: The method is with @Transactional tag, so if a keycloak returns an error the data won't be deleted, if the delete method returns an error the keycloak code won't be executed

**Pull request merged**